src/test/script_segwit_tests.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2012-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <script/script.h>
+#include <test/test_bitcoin.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(script_segwit_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(IsPayToWitnessScriptHash_Valid)
+{
+    uint256 dummy;
+    CScript p2wsh;
+    p2wsh << OP_0 << ToByteVector(dummy);
+    BOOST_CHECK(p2wsh.IsPayToWitnessScriptHash());
+
+    std::vector<unsigned char> bytes = {OP_0, 32};
+    bytes.insert(bytes.end(), 32, 0);
+    BOOST_CHECK(CScript(bytes.begin(), bytes.end()).IsPayToWitnessScriptHash());
+}
+
+BOOST_AUTO_TEST_CASE(IsPayToWitnessScriptHash_Invalid_NotOp0)
+{
+    uint256 dummy;
+    CScript notp2wsh;
+    notp2wsh << OP_1 << ToByteVector(dummy);
+    BOOST_CHECK(!notp2wsh.IsPayToWitnessScriptHash ());
```

Comment:
Nit: Remove space before `()` :-)

Applies throughout this PR.

Reply:
Good catch (that's left over from me coding GNU style most of the time outside of Bitcoin), fixed.